core-addon/IonCore.js

```diff
@@ -222,6 +222,15 @@ module.exports = class IonCore {
     }
 
     switch (message.type) {
+      case "core-check": {
+        let enrolled = !!(await this._storage.getIonID());
+        return Promise.resolve({
```

Comment:
nit, does returning `Promise.resolve()` from an `async` function do anything different than just `return`? I think async/await just does syntactic sugar over this and auto-wrap/unwrap of promises so I don't see the difference testing this out locally.

Reply:
You're right! Fixing this.